### PR TITLE
[AAP-57909] Fix for the custom role team assignment

### PR DESCRIPTION
**Pull request by nickbhasin**
## Description

Assigning a custom role to the team is getting ignored.
- What is being changed?
Entity type was being assigned incorrectly so that it only works for inbuilt roles.
- Why is this change needed?
To allow users to assign a custom role to the team.
- How does this change address the issue?
Now entity type relies on the actual API metadata of the role.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

### Steps to Test
1. 
2. 
3. 

### Required Actions

- [ ] Requires documentation updates
  
- [ ] Requires downstream repository changes
  
- [ ] Requires infrastructure/deployment changes
  
- [ ] Requires coordination with other teams
  
- [ ] Blocked by PR/MR: #XXX

### CODE REVIEW

**Review by rohitthakur2590 — APPROVED** [review]

File: tests/integration/targets/role_team_assignments_test/tasks/main.yml [hunk 1]
```diff
@@ -113,6 +128,23 @@
...
+        role_definition: "{{ custom_role_name }}"
+        team: "{{ team1.name }}"
+        state: present
+      register: custom_role_assignment
+
+    - name: Assert Custom Role Assignment was successful
+      ansible.builtin.assert:
+        that:
+          - custom_role_assignment is changed
+          - custom_role_assignment.assignments | length > 0
```
Comment (rohitthakur2590) [inline tests/integration/targets/role_team_assignments_test/tasks/main.yml]: can we do query to assert this ?

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: tests/integration/targets/role_team_assignments_test/tasks/main.yml (hunk 1 above)
Comment (nickbhasin) [reply tests/integration/targets/role_team_assignments_test/tasks/main.yml]: done.

**Review by rohitthakur2590 — APPROVED** [review]
